src/ciceroscm/carbon_cycle/carbon_cycle_mod.py

```diff
@@ -229,14 +264,95 @@ def precalc_r_functions(self):
                 idtm=self.pamset["idtm"],
             )
 
-    def _calculate_partial_pressure_mixed_layer(self, it):
+    def fnpp_from_temp(self, dtemp=0):
+        """
+        Temperature dependence function for fnpp
+
+        Parameters
+        ----------
+        dtemp : float
+            Degrees of temperature since start of run
+
+        Returns
+        -------
+        float
+            fnpp at given temperature for assumed linear
+            relationship
+        """
+        sigmoid = partial(
+            sigmoid_gen,
+            sigmoid_center=self.pamset["t_half"],
+            sigmoid_width=self.pamset["w_sigmoid"],
+        )
+        threshold = partial(
+            threshold_gen,
+            threshold_half=self.pamset["t_threshold"],
+            threshold_width=self.pamset["w_threshold"],
+        )
+        # Threshold function
+        baseline = self.pamset["npp0"] / (sigmoid(0) * threshold(0))
+
+        return baseline * sigmoid(dtemp) * threshold(dtemp)
+
+    def solubility_temp_feedback(self, dtemp=0.0):
+        """
+        Exponential scaling of CO2 solubility with temperature, with upper limit.
+
+        The parameters used are solubility_sens, the fractional decrease in solubility
+        per degree C (default 0.02) and solubility_limit, the maximum allowed gain
+        (e.g. 0.5 for 50% increase)
+
+        Parameters
+        ----------
+        dtemp : float
+            Degrees of temperature since start of run
+
+
+        Returns
+        -------
+        float
+            Solubility scaling factor (max 1 + solubility_limit)
+        """
+        scale = np.exp(-self.pamset["solubility_sens"] * dtemp) / np.exp(
+            -self.pamset["solubility_sens"] * 0.0
```

Comment:
Can I just delete the overcomplicated divide by 1 here, or is this a relic of something that was originally meaningful which disappeared when code moved around?